types/src/on_chain_config.rs

```diff
@@ -19,31 +58,23 @@ pub trait ConfigStorage {
 
 /// Trait to be implemented by a Rust struct representation of an on-chain config
 /// that is stored in storage as a deserialized byte array
-pub trait OnChainConfig {
+pub trait OnChainConfig: Send + Sync {
     const IDENTIFIER: &'static str;
+    const CONFIG_ID: ConfigID = ConfigID(Self::IDENTIFIER);
 
-    fn deserialize_into_config(bytes: &[u8]) -> Result<Self>
+    fn deserialize_default_impl(bytes: &[u8]) -> Result<Self>
     where
         Self: DeserializeOwned,
     {
-        let result = lcs::from_bytes::<Self>(&bytes).map_err(|e| {
-            anyhow::format_err!(
-                "Failed to serialize account blob bytes to on-chain config: {}",
-                e
-            )
-        })?;
-        Ok(result)
+        lcs::from_bytes::<Self>(&bytes)
+            .map_err(|e| format_err!("[on-chain config] Failed to deserialize into config: {}", e))
     }
 
-    fn fetch_config<U: ConfigStorage>(storage: &U) -> Result<Self>
+    fn deserialize_into_config(bytes: &[u8]) -> Result<Self>
```

Comment:
hmm, can we add some comments about the two functions? I'm not sure I understand the semantics

Reply:
will do in code, but also can comment here as well: this functionality is pretty common for when overriding `deserialize_into_config` in implementation (see `VMPublishingOption` - deserializing is a bit different there) (fyi we can't call `OnChainConfig::deserialize_into_config` in the override - it will just call the overriden method itself)